benchmarks/benchmark_shared.c

```diff
@@ -116,3 +120,61 @@ int pingpong(void)
 
 	return 0;
 }
+
+int bandwidth(void)
+{
+	int ret, i, j;
+
+	ret = ft_sync();
+	if (ret)
+		return ret;
+
+	if (opts.dst_addr) {
+		for (i = 0; i < opts.iterations + opts.warmup_iterations; i++) {
+			if (i == opts.warmup_iterations)
+				ft_start();
+
+			for(j = 0; j < opts.window_size; j++) {
+				if (opts.transfer_size < fi->tx_attr->inject_size)
+					ret = ft_inject(opts.transfer_size);
+				else
+					ret = ft_post_tx(opts.transfer_size);
+				if (ret)
+					return ret;
+			}
+			ret = ft_get_tx_comp(tx_seq);
+			if (ret)
+				return ret;
+			ret = ft_rx(4);
+			if (ret)
+				return ret;
+		}
+	} else {
+		for (i = 0; i < opts.iterations + opts.warmup_iterations; i++) {
+			if (i == opts.warmup_iterations)
+				ft_start();
+
+			for(j = 0; j < opts.window_size; j++) {
+				ret = ft_post_rx(opts.transfer_size);
+				if (ret)
+					return ret;
+			}
```

Comment:
This allows for the possibility that the sender immediately overruns the receiving side on the first transfer, which could either results in buffering or the sender backing off some time and retrying.  In either case, the performance could be highly variable.